### ISSUE TITLE
Allow Topology to specify whether SystemGenerator.create_system() creates periodic or non-periodic systems

### ISSUE THREAD

**Issue by jchodera** [openmm/openmm-forcefields#92]
Currently, `SystemGenerator` accepts some `forcefield_kwargs` that are generally used to specify `nonbondedMethod`. Despite this, the `Topology` object can be either periodic (contains box vectors) or non-periodic (no box vectors), which can cause problems since the `nonbondedMethod` specifies either a periodic or non-periodic system.

Instead, we could just detect from `Topology` whether a periodic or non-periodic system is to be generated, choosing an appropriate `nonbondedMethod` for the appropriate system. 

Some possibilities:
1. Split `forcefield_kwargs` to `periodic_kwargs` and `nonperiodic_kwargs`:
```python
system_generator = SystemGenerator(forcefields=forcefields, 
    small_molecule_forcefield='openff-1.0.0', 
    periodic_kwargs={'removeCMMotion': False, 'nonbondedMethod': app.PME, 'constraints' : app.HBonds, 'hydrogenMass' : 4*unit.amu},
    nonperiodic_kwargs={'removeCMMotion': False, 'nonbondedMethod': app.NoCutoff, 'constraints' : app.HBonds, 'hydrogenMass' : 4*unit.amu})
```
2. Add `periodic_kwargs` and `nonperiodic_kwargs` to be used in addition to `forcefield_kwargs` 
```python
system_generator = SystemGenerator(forcefields=forcefields, 
    small_molecule_forcefield='openff-1.0.0', 
    forcefield_kwargs={'removeCMMotion': False, 'constraints' : app.HBonds, 'hydrogenMass' : 4*unit.amu},
    periodic_kwargs={'nonbondedMethod': app.PME},
    nonperiodic_kwargs={'nonbondedMethod': app.NoCutoff})
```
3. Simply have `periodic_nonbonded_method` and `nonperiodic_nonbonded_method` that are added to `forcefield_kwargs`
```python
system_generator = SystemGenerator(forcefields=forcefields, 
    small_molecule_forcefield='openff-1.0.0', 
    forcefield_kwargs={'removeCMMotion': False, 'constraints' : app.HBonds, 'hydrogenMass' : 4*unit.amu},
    periodic_nonbonded_method=app.PME,
    nonperiodic_nonbonded_method=app.NoCutoff)
```
Any thoughts @hannahbrucemacdonald @dominicrufa?

**Comment by hannahbrucemacdonald** [comment]
I'm happy with any, slight preference for 3

**Comment by dominicrufa** [comment]
if we don't foresee any other kwargs for periodics and nonperiodics, then i have a preference for 3

**Comment by jchodera** [comment]
You folks are great---thanks for the quick feedback. I'll implement Option 3 right now!

**Comment by jchodera** [comment]
Actually, I worry that this may not be as general when, for example, polarizable force fields are in use. Option 2 is still convenient but much more general.

**Comment by jchodera** [comment]
Addressed by #93
